pkg/ccl/changefeedccl/alter_changefeed_stmt.go

```diff
@@ -308,38 +342,53 @@ func generateNewTargets(
 	// perform these validations in the validateNewTargets function.
 	allDescs, err := backupresolver.LoadAllDescs(ctx, p.ExecCfg(), statementTime)
 	if err != nil {
-		return nil, nil, hlc.Timestamp{}, err
+		return nil, nil, hlc.Timestamp{}, nil, err
 	}
 	descResolver, err := backupresolver.NewDescriptorResolver(allDescs)
 	if err != nil {
-		return nil, nil, hlc.Timestamp{}, err
+		return nil, nil, hlc.Timestamp{}, nil, err
 	}
 
-	for _, target := range AllTargets(prevDetails) {
-		k := targetKey{TableID: target.TableID, FamilyName: tree.Name(target.FamilyName)}
-		newTargets[k] = tree.ChangefeedTarget{
-			TableName:  tree.NewUnresolvedName(target.StatementTimeName),
-			FamilyName: tree.Name(target.FamilyName),
+	for _, targetSpec := range AllTargets(prevDetails) {
+		k := targetKey{TableID: targetSpec.TableID, FamilyName: tree.Name(targetSpec.FamilyName)}
+		desc := descResolver.DescByID[targetSpec.TableID].(catalog.TableDescriptor)
+
+		tbName, err := getQualifiedTableNameObj(ctx, p.ExecCfg(), p.ExtendedEvalContext().Txn, desc)
```

Comment:
Why do we need to resolve anything in the prevDetails? Why can't we just use prevDetails.TargetSpecifications?

Reply:
It also appears to me that perhaps having a tableID -> target ChangefeedTargetSpecification is a better structure?

Reply:
> It also appears to me that perhaps having a tableID -> target ChangefeedTargetSpecification is a better structure?

Since column families would have the same tableID, but different statement time names, I'm not sure if this mapping would work.

> Why do we need to resolve anything in the prevDetails? Why can't we just use prevDetails.TargetSpecifications?

We can't just use the prevDetails.TargetSpecifications because we would have no way of knowing which target specification to use in prevDetails (if any) for any ChangefeedTarget in `rawTargets` in the function `getTargetsAndTables`. Recall that targets can also get dropped, so we can't use prevDetails.TargetSpecifications in that case.

If we wanted to go down the route of just using prevDetails.TargetSpecifications, we would also need a way to find out which targets in `rawTargets` are going to use prevDetails and which ones will not. We could set up `originalSpecs` so that it includes all of the TargetSpecifications that we would need (so `changefeed_stmt` would not even need to call `getTargetsAndTables`), but then we would be performing a lot of duplicate work that is in the function `getTargetsAndTables`.

Another nice added bonus of performing a resolve is that we can update the changefeed description to the new target name in the event of a table rename.

Reply:
@HonoreDB might want to speak more on these questions